### ISSUE TITLE
Skipped interval length creates microtask instead of task

### ISSUE THREAD

**Issue by artalar** [oven-sh/bun#2121]
### What version of Bun is running?

0.5.6

### What platform is your computer?

Linux 5.15.63 x86_64 x86_64

### What steps can reproduce the bug?

```ts
setTimeout(() => console.log(2));
setTimeout(() => console.log(3), 0);
setTimeout(() => console.log(4), 1);
Promise.resolve().then(() => console.log(1));
```

https://codesandbox.io/p/sandbox/magical-gould-nnyxtu

### What is the expected behavior?

1
2
3
4

### What do you see instead?

2
1
3
4

### Additional information

How did I find it?

https://github.com/artalar/reatom/blob/a051f142f77c51551711124c2d638b31cd40055f/packages/core/src/atom.test.ts

<img width="1048" alt="CleanShot 2023-02-21 at 15 37 07@2x" src="https://user-images.githubusercontent.com/27290320/220346848-7c30b181-11e8-4328-994b-13699ab5d04d.png">

**Comment by Jarred-Sumner** [comment]
Fixed in runtime code, but the test runner still reports this. Likely an issue with some of the event loop code that impacts the test runner

**Comment by Electroid** [comment]
Fixed as of Bun v1.0.26.

```
❯ bun index.test.ts
1
2
3
4
```
```
❯ bun test index.test.ts
bun test v1.0.26 (1f0c1c81)

index.test.ts:
1
2
3
4
✓ works [1001.27ms]

 1 pass
 0 fail
Ran 1 tests across 1 files. [1021.00ms]
```
